### PR TITLE
Extract and unit test StatefulSets upscale

### PR DESCRIPTION
**Pull request by sebgl**
Extract smaller functions to build expected resources and upscale/apply
StatefulSets configuration, and add unit tests for both.

### CODE REVIEW

File: operators/pkg/controller/elasticsearch/driver/nodes.go [hunk 1]
```diff
@@ -143,3 +108,26 @@ func (d *defaultDriver) reconcileNodeSpecs(
 	//  - grow and shrink
 	return results
 }
+
+func expectedResources(
```
Comment (barkbay) [inline operators/pkg/controller/elasticsearch/driver/nodes.go]: nit: I would not expect a function called `expectedResources` to have a side effect on the cluster

File: operators/pkg/controller/elasticsearch/driver/nodes_test.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+					NodeCount: 3,
+				},
+			},
+		},
+	}
+
+	resources, err := expectedResources(k8s.WrapClient(fake.NewFakeClient()), es, observer.State{}, nil)
+	require.NoError(t, err)
+
+	// 3 StatefulSets should be created
```
Comment (barkbay) [inline operators/pkg/controller/elasticsearch/driver/nodes_test.go]: 2 ?

File: operators/pkg/controller/elasticsearch/driver/nodes_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
```
Comment (barkbay) [inline operators/pkg/controller/elasticsearch/driver/nodes_test.go]: same here ?

File: operators/pkg/controller/elasticsearch/driver/nodes.go (hunk 1 above)
Comment (sebgl) [reply operators/pkg/controller/elasticsearch/driver/nodes.go]: I don't think there is a side effect on the cluster?
I guess what you refer to is that we patch the config of the expected resources for zen1/zen2, but those are just the expected resources in memory, we don't change anything in the apiserver here, nor make any ES request.

**Pull request merged**